### PR TITLE
upgrade(Node): Updage Node Version 14.x -> 18.x

### PR DESCRIPTION
**Pull request by kotva006**
#### Summary of changes
No Ticket.
One of the linting dependencies was failing because of an outdated node version.
This upgrade fixes the main linting problem, and updates some tests to work with the new node library.

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

File: apps/site/assets/ts/schedule/components/__tests__/ScheduleLoaderTest.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
 import { Provider } from "react-redux";
+import "whatwg-fetch";
```
Comment (thecristen) [inline apps/site/assets/ts/schedule/components/__tests__/ScheduleLoaderTest.tsx:3]: Might it work to do this in our Jest setup file?

File: apps/site/assets/ts/schedule/components/schedule-finder/__tests__/ScheduleModalContentTest.tsx [hunk 2]
```diff
@@ -74,9 +75,10 @@ describe("ScheduleModalContent", () => {
       () =>
         new Promise((resolve: Function) =>
           resolve({
-            json: () => "Internal Server Error",
-            ok: false,
-            status: 200
+            json: () => [{ trip: { id: "yeah" } }],
```
Comment (thecristen) [inline apps/site/assets/ts/schedule/components/schedule-finder/__tests__/ScheduleModalContentTest.tsx:78]: Yep. 😆

File: apps/site/assets/ts/schedule/components/schedule-finder/ScheduleModalContent.tsx [hunk 3]
```diff
@@ -36,6 +36,8 @@ export const fetchData = async (
     `/schedules/finder_api/departures?id=${routeId}&stop=${selectedOrigin}&direction=${selectedDirection}`
   );
 
+  console.log(departures);
+
```
Comment (thecristen) [inline apps/site/assets/ts/schedule/components/schedule-finder/ScheduleModalContent.tsx:40]: found the one linting error! please remove that.

**Review by thecristen — CHANGES_REQUESTED** [review]
🤩 looking promising, just some minor comments

File: apps/site/assets/ts/schedule/components/__tests__/ScheduleLoaderTest.tsx (hunk 1 above)
Comment (kotva006) [reply apps/site/assets/ts/schedule/components/__tests__/ScheduleLoaderTest.tsx:3]: Good call, forgot this was a file

**Review by thecristen — APPROVED** [review]

**Comment by thecristen** [comment]
@kotva006 Thanks!!!!

**Pull request merged**